sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -234,28 +234,30 @@ public DirectConnectionConfig setMaxRequestsPerConnection(int maxRequestsPerConn
     }
 
     /**
-     * Gets the request timeout interval
-     * This represents the timeout interval for requests
+     * Gets the network request timeout interval
+     * This represents the time to wait for response from network peer
      *
-     * Default value is 60 seconds
+     * Default value is 5 seconds
      *
-     * @return the request timeout interval
+     * @return the network request timeout interval
      */
-    Duration getRequestTimeout() {
-        return requestTimeout;
+    public Duration getNetworkRequestTimeout() {
+        return networkRequestTimeout;
     }
 
     /**
-     * Sets the request timeout interval
-     * This represents the timeout interval for requests
+     * Sets the network request timeout interval
+     * This represents the time to wait for response from network peer
      *
-     * Default value is 5 seconds
+     * Default value is 5 seconds.
+     * This value will impact the number of times retries happen within SDK.
+     * We strongly recommend setting the value less than 10s.
      *
-     * @param requestTimeout the request timeout interval
+     * @param networkRequestTimeout the network request timeout interval
      * @return the {@link DirectConnectionConfig}
      */
-    DirectConnectionConfig setRequestTimeout(Duration requestTimeout) {
-        this.requestTimeout = requestTimeout;
+    public DirectConnectionConfig setNetworkRequestTimeout(Duration networkRequestTimeout) {
```

Comment:
the other timeout values we have in DirectConnectionConfig are also network timeout but they don't have network prefix. e.g. idleConnectionTimeout, or connectTimeout.

I understand why we added network as prefix to disambiguate from e2e timeout, could you pleasecheck how we did this disambiguation on the dotnet side?

Reply:
@j82w what is this config called on the DotNet side?

Reply:
CosmosClientOptions.RequestTimeout
https://github.com/Azure/azure-cosmos-dotnet-v3/blob/1b6fa9da0289f4cf447eef7dc28652e09f0dfec1/Microsoft.Azure.Cosmos/src/CosmosClientOptions.cs#L161

Reply:
The current requestTimeout is not e2e requestTimeout right? 
If in the future, we are going to support e2e requestTimeout, what we plan to do?
Reusing requestTimeout, and it means e2e. How customer can config the requestTimeout per network call then?

or 

having a new config for e2e?

Reply:
The .NET SDK does not have a e2e timeout. They can use the cancellation token and RequestTimeout which represent the Networking timeout to get a e2e timeout.

Reply:
got it.

In .net, are we planning to exposing a e2e RequestTimeout API? or cancellationToken + requestTimeout (network) will be enough?

Reply:
CancellationToken is not a timeout on the sense that it stops exactly when the user wants, but it's the approach that .NET has for async operations. The CT is checked whenever it is safe to do so and won't cause an invalid state. This means that the check can be done some time after the CT has actually expired because that was when it was safe.

Reply:
We check it across retries internally, so if they are getting 410s but the CT expired, then the operation is halted.

Reply:
Based on the discussion on the team meeting: Going to keep using networkRequestTime for now. In java, there might be a need to expose a e2e requestTimeout in the future.